File: Modulo_3/PHP/ejercicios.php
<?php

//Ejercicio 1

//Se declaran los contadores
$pares = 0;
$impares = 0;

//Se recorren los numeros del 1 al 50
for ($i = 1; $i <= 50; $i++) {
    if ($i % 2 == 0) { //Si es divisible entre 2, es par
        $pares++;
    } else { //Si no, es impar
        $impares++;
    }
}

//Se muestran los resultados
echo "Cantidad de numeros pares entre 1 y 50: " . $pares . "";
echo "\n";
echo "Cantidad de numeros impares entre 1 y 50: " . $impares . "";
echo "\n";

//Ejercicio 2
//En este ejercicio, en vez de limitarme al numero 8, simplemente coloque un readline para asi poder ver la tabla de multiplicar del numero que quiera

$numero = readline("\nIngrese un numero: ");  

echo "\nTabla de multiplicar del " . $numero . ":\n";

//Generamos la tabla del multiplicar del numero ingresado
for ($i = 1; $i <= 10; $i++) {
    echo $numero . " x " . $i . " = " . ($numero * $i) . "\n";
}
echo "\n";

//Ejercicio 3

$num_secreto = rand(1,10); //Se genera un numero aleatorio del 1 al 10

echo "Adivina el numero (entre 1 y 10):\n";

while ((int)trim(fgets(STDIN)) != $num_secreto) { //Empieza un bucle while
  echo "Incorrecto. Intenta otra vez ";
}

echo "Lo adivinaste!\n"; //El bucle termina una vez que adivinado el numero generado

//Ejercicio 4

$suma = 0; //Declaramos la variable para acumular la suma

//Recorremos los numeoros del 1 al 100
for ($i = 1; $i <= 100; $i++) {
  if ($i % 2 != 0) { //Si es impar, los sumamos
    $suma += $i;
  }
}

//Mostramos el resultado
echo "\nLa suma de todos los numeros impares del 1 al 100 es: " . $suma . "";
echo "\n";

//Ejercicio 5

$edad = readline("\nIntroduce tu edad: "); //Se usa en readline para introducir la edad

echo "Verificacion de edad para licencia de conducir:\n";
echo "Edad ingresada: " . $edad . " años\n";

//Se verifican los requisitos
if ($edad >= 18 && $edad <= 65) {
  echo "Cumple con los requisitos para obtener la licencia.\n"; 
} else {
    if ($edad < 18) {
    echo "No cumple con los requisitos para obtener la licencia.  Motivo: Es menor de 18 años\n";
    } else {
        echo "No cumple con los requisitos. Motivo: Supera el limite de 65 años\n";
    }   
    echo "\n";
}


//Ejercicio 6 

$tamaño = 5; //Se define el tamaño del cuadrado (5x5)

//Bucle externo para las filas 
for ($i = 1; $i <= $tamaño; $i++) {
    //Bucle interno para las columnas
  for ($l = 1; $l <= $tamaño; $l++) {
    echo "# "; //Se imprime el cuadro
  }
  echo "\n";
}

//Ejercicio 7

$num = readline("\nIngrese un numero: "); //Se utiliza de nuevo el readline para introducir un numero

//Condicionales para la clasificacion
if ($num > 0) {
  echo "El numero es positivo\n";
} elseif ($num < 0) {
  echo "El numero es negativo\n";
} else {
  echo "El numero es cero\n";
} 


//Ejercicio 8

echo "\n";
//Bucle para recorrer los numeros del 1 al 30
for ($i = 1; $i <= 30; $i++) {
    //Posteriormente, se verifican todas las condiciones
    if ($i % 3 == 0 && $i % 5 == 0) {
        echo "MarTierra ";
    } elseif ($i % 3 == 0) {
        echo "Mar ";
    } elseif ($i % 5 == 0) {
        echo "Tierra ";
    } else { //Si no se cumple ninguna condicion anterior
        echo $i . " ";
    }
    echo "\n";
}

//Ejercicio 9

$temperatura = readline("\nIngrese la temperatura: "); //Otro readline mas para introducir la temperatura 

//Condicionales para la clasificacion
if ($temperatura < 10) {
  echo "Clima frio\n";
} elseif ($temperatura >= 10 && $temperatura <= 25) {
  echo "Clima templado\n";
} else {
  echo "Clima Caluroso\n";
}

//Ejercicio 10

echo "\nCuenta regresiva para Año Nuevo:\n";

//Bucle descendente del 10 al 1
for ($i = 10; $i >= 1; $i--) {
    echo $i . "...\n";
}

//Mensaje final
echo "Feliz Año Nuevo! \n";
?>

File: Modulo_3/C++/Ejercicios.cpp
#include <iostream>
#include <cmath>
using namespace std;

//Ejercicio 1

int main() {
    cout << "¡Hola Mundo!" << endl;

    int edad;
    cout << "Introduce tu edad: ";
    cin >> edad;

    cout << "Tienes " << edad << " años." << endl;

    float x = 9.25;
    float y = 3.26;

    int multiplicacion1 = x * y;
    cout << "x * y es igual a: " << multiplicacion1 << endl;

    char letra = 'H';
    cout << "Char: " << letra << endl;
    char letra2 = 'O';
    cout << "Char: " << letra2 << endl;
    char letra3 = 'L';
    cout << "Char: " << letra3 << endl;
    char letra4 = 'A';
    cout << "Char: " << letra4 << endl;

    int numero;

    cout << "Ingrese un numero: ";
    cin >> numero;

    bool resultado = (numero % 2 == 0);

    cout << "¿Es par? " << resultado << endl;

    if (resultado) {
        cout << numero << " es par" << endl;
    } else {
        cout << numero << " es impar" << endl;
    }

//Ejercicio 2

    int num1;
    cout << "Ingrese el primer numero: ";
    cin >> num1;

    int num2;
    cout << "Ingrese el segundo numero: ";
    cin >> num2;

    int suma = num1 + num2;
    cout << num1 << " + " << num2 << " es igual a " << suma << endl;
    int resta = num1 - num2;
    cout << num1 << " - " << num2 << " es igual a " << resta << endl;
    int multiplicacion2 = num1 * num2;
    cout << num1 << " * " << num2 << " es igual a " << multiplicacion2 << endl;
    int division = num1 / num2;
    cout << num1 << " / " << num2 << " es igual a " << division << endl;
    int raiz = sqrt(num1);
    cout << "La raíz cuadrada de " << num1 << " es " << raiz << endl;
    int potencia = pow(num1, num2);
    cout << num1 << " elevado a " << num2 << " es " << potencia << endl;

//Ejercicio 3

    int edad2;
    cout << "Introduce tu edad: ";
    cin >> edad2;

    if (edad2 >= 18) {
        cout << "Eres mayor de edad." << endl;
    } else {
        cout << "Eres menor de edad." << endl;
    }

//Ejercicio 4

    #define LIMITE 10

    int num;
    cout << "Introduce un numero: ";
    cin >> num;
    
    cout << "Tabla de multiplicar del: " << num << endl;

    for (int i = 1; i <= 10; i++) {
        cout << num << " x " << i << " = " << (num * i) << endl;
    }

    return 0;
}